src/vivarium_gates_nutrition_optimization_child/components/observers.py

```diff
@@ -308,3 +308,33 @@ def setup(self, builder: Builder) -> None:
         self.entity_type = self.disease_model.cause_type
         self.entity = self.disease_model.cause
         self.transition_stratification_name = f"transition_{self.disease}"
+
+    # We want disease state to be categories instead of the (default) state_ids
+    def register_disease_state_stratification(self, builder: Builder) -> None:
+        builder.results.register_stratification(
+            "wasting_categories",
+            list(builder.data.load(f"risk_factor.{self.disease}.categories").keys()),
```

Comment:
I think you can also get the categories from WASTING.CATEGORIES in constants/data_values.py.

Reply:
Ah, interesting. I just copied what had been used previously - it does seem like there are lots of duplicate ways of doing the same thing in this model (probably from handing off between engineers?)